### PR TITLE
Updated TOC to go across all 3 Parts

### PR DESCRIPTION
**Pull request by andyjohnson**
Updated TOC to go across all 3 Parts - eventually will split into 3
documents after all links are done and the current "gray/coding" area
has a permanent home.

### CODE REVIEW

**Comment by rswetnam** [comment]
I don't see the merge as having been made - but certainly support turning this pull request into a merge asap.

File: xAPI.md [hunk 1]
```diff
@@ -186,59 +186,70 @@ learning that SCORM could not enable.
...
-    *	7.11.	[HTTP HEAD](#httphead)  
+*	Part One:	[Summary of the Experience API](#partone)  
+	*	1.0.	[Role of the Experience API](#roleofxapi) 
+	*	2.0.	[How To Use This Document](#readingguidelines)  
+		*	2.1.	[MUST / SHOULD / MAY](#def-must-should-may)  
+	 	*	2.2.	[Guidelines for Interpreting Descriptive Text and Tables](#interpret-text-table)  
+	*	3.0.	[Serialization and JavaScript Object Notation](#json)
+	*	4.0.	[Definitions](#definitions) 
+	*	5.0.	[xAPI Components](#extending-xapi) 
+	*	6.0.	[Extending xAPI](#roleofxapi)
```
Comment (garemoko) [inline xAPI.md]: Lines 196 and 197 have incorrect links.

File: xAPI.md [hunk 2]
```diff
@@ -186,59 +186,70 @@ learning that SCORM could not enable.
...
+		*	2.1.	[MUST / SHOULD / MAY](#def-must-should-may)  
+	 	*	2.2.	[Guidelines for Interpreting Descriptive Text and Tables](#interpret-text-table)  
+	*	3.0.	[Serialization and JavaScript Object Notation](#json)
+	*	4.0.	[Definitions](#definitions) 
+	*	5.0.	[xAPI Components](#extending-xapi) 
+	*	6.0.	[Extending xAPI](#roleofxapi) 
+	*	7.0.	[Profiles and Communities of Practice](#COPs)  
+*	Part Two:	[Experience API (xAPI) Data](#parttwo)  
+	*	1.0.	[Documents](#documents) 
+	*	2.0.	[Statements](#statement)
```
Comment (garemoko) [inline xAPI.md]: Links to the appendix

File: xAPI.md [hunk 3]
```diff
@@ -186,59 +186,70 @@ learning that SCORM could not enable.
...
+	*	6.0.	[Extending xAPI](#roleofxapi) 
+	*	7.0.	[Profiles and Communities of Practice](#COPs)  
+*	Part Two:	[Experience API (xAPI) Data](#parttwo)  
+	*	1.0.	[Documents](#documents) 
+	*	2.0.	[Statements](#statement)  
+		*	2.1.	[Purpose](#statement-purpose)  
+	 	*	2.2.	[Formatting Requirements](#dataconstraints) 
+	 	*	2.3.	[Statement Lifecycle](#lifecycle) 
+		 	*	2.3.1.	[Statement Immutability](#statement-immutablity-and-exceptions) 
+		 	*	2.3.2.	[Voiding](#voided)
```
Comment (garemoko) [inline xAPI.md]: Links to wrong place (maybe we have an extra anchor in the middle of statement lifecycle?)

File: xAPI.md [hunk 4]
```diff
@@ -186,59 +186,70 @@ learning that SCORM could not enable.
...
+	*	7.0.	[Profiles and Communities of Practice](#COPs)  
+*	Part Two:	[Experience API (xAPI) Data](#parttwo)  
+	*	1.0.	[Documents](#documents) 
+	*	2.0.	[Statements](#statement)  
+		*	2.1.	[Purpose](#statement-purpose)  
+	 	*	2.2.	[Formatting Requirements](#dataconstraints) 
+	 	*	2.3.	[Statement Lifecycle](#lifecycle) 
+		 	*	2.3.1.	[Statement Immutability](#statement-immutablity-and-exceptions) 
+		 	*	2.3.2.	[Voiding](#voided) 
+   		*	2.4.	[Statement Properties](#stmtprops)
```
Comment (garemoko) [inline xAPI.md]: Doesn't go anywhere

**Comment by garemoko** [comment]
There's 5 errors to fix. Other than those, +1.

File: xAPI.md (hunk 2 above)
Comment (garemoko) [inline xAPI.md]: This is also broken in 1.0.2, but worth fixing in this PR.

**Comment by andyjohnson** [comment]
Thanks @garemoko !  Should be fixed now.  Anyone else with a +1 on this?

**Comment by rswetnam** [comment]
For me, a Table of Contents has 2 primary functions:
1. To provide an overview of the document and a glimpse into the way that the author's mind works;
2. To help the reader navigate the document more easily.

For that reason, I suggest removing removing level 3 headings in the TOC such as Statement Properties.  Here items such as actor, verb, etc. are parts of a local table.  I suggest removing references to these items in the TOC and going through the document and making sure that all uses of these terms are hyperlinked to the definitions.  This will not only help navigation but it will help make the document organization more evident.

**Comment by rswetnam** [comment]
That being said, I support the TOC as they now stand being merged.  +1 from me.

File: xAPI.md [hunk 5]
```diff
@@ -186,68 +186,80 @@ learning that SCORM could not enable.
...
+		*	2.1.	[MUST / SHOULD / MAY](#def-must-should-may)  
+	 	*	2.2.	[Guidelines for Interpreting Descriptive Text and Tables](#interpret-text-table)  
+	*	3.0.	[Serialization and JavaScript Object Notation](#json)
+	*	4.0.	[Definitions](#definitions) 
+	*	5.0.	[xAPI Components](#xapi-components) 
+	*	6.0.	[Extending xAPI](#extending-xapi) 
+	*	7.0.	[Profiles and Communities of Practice](#COPs)  
+*	Part Two:	[Experience API (xAPI) Data](#parttwo)  
+	*	1.0.	[Documents](#documents) 
+	*	2.0.	[Statements](#statement)
```
Comment (garemoko) [inline xAPI.md]: Sorry, my last comment here was unclear. Whilst the improvements you've made to the appendices links seem good, what I meant was that when I click this statements link I end up in the appendix.

**Comment by garemoko** [comment]
@rswetnam - I regularly use those third tier links in the contents. I find them really helpful for looking up a particular statement property.

**Comment by andyjohnson** [comment]
Ahh, I got it!   For some reason the "statement" property in the example was triggering the anchor.  Changed it and the 2.0 Section anchor to "statements"

**Pull request merged**